plainkey/platform.go

```diff
@@ -0,0 +1,156 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package plainkey
+
+import (
+	"bytes"
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/hmac"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sync"
+
+	"golang.org/x/crypto/cryptobyte"
+
+	"github.com/snapcore/secboot"
+)
+
+const (
+	platformName = "plainkey"
+)
+
+var (
+	platformKeysMu sync.RWMutex
+	platformKeys   [][]byte
+)
+
+// SetPlatformKeys sets the keys that will be used by this platform to recover other
+// keys. These are typically stored in and loaded from an encrypted container that is
+// unlocked via some other mechanism.
+func SetPlatformKeys(keys ...[]byte) {
+	platformKeysMu.Lock()
+	platformKeys = keys
+	platformKeysMu.Unlock()
+}
+
+// AddPlatformKeys adds keys that will be used by this platform to recover other
+// keys. These are typically stored in and loaded from an encrypted container that is
+// unlocked via some other mechanism.
+func AddPlatformKeys(keys ...[]byte) {
```

Comment:
do we need this? anyway it seems not to be tested atm

Reply:
I've removed this.